src/ai/composite/goal.cpp

```diff
@@ -313,17 +313,13 @@ void lua_goal::add_targets(std::back_insert_iterator< std::vector< target >> tar
 	config c(cfg_.child_or_empty("args"));
 	const config empty_cfg;
 	handler_->handle(c, empty_cfg, true, l_obj);
-	try {
-		std::vector < target > targets = *(l_obj->get());
 
-		for (target tg : targets)
-		{
-			*target_list = tg;
-		}
-	} catch(const bad_enum_cast& e) {
-		ERR_AI_GOAL << "A Lua goal returned a target of an unknown type (\"" << e.value() << "\"; unfortunately, the engine cannot recover from this error. As a result, all targets returned by the goal have been lost.\n";
-	}
+	std::vector < target > targets = *(l_obj->get());
```

Comment:
Is there any way to get the error message back? Obviously the exact details are incorrect (even before, it looks like it managed to partially recover), but I think at least noting when there's an invalid target type is a good idea (and maybe don't insert the invalid type into the list at all).

Reply:
I removed it since I didn't see what would have resulted in a bad_enum_cast here. `target` already has an enum member variable, so it didn't look like there was a place where it was converting from a string.

Reply:
It definitely exists somewhere, because the Lua API returns the target type as a string (or an integer, which I think is deprecated).

Reply:
I'm assuming the actual reading of the value happens at the point in lua_object.hpp: https://github.com/wesnoth/wesnoth/pull/6470/commits/f0b90963c4cf6cde8d21fd7f6101156056b1de84#diff-cf0d01f4533748ed6e5ea6b39b78a6c4a4d3cd7513a9b806094c6d527dc14ac1 so I can change the bad value handling and/or add logging there.

Reply:
So instead of using `value_or()`, it should check if it converted correctly and if not then skip it with `continue`?

Reply:
Something like that seems reasonable, yeah.

Reply:
Alright, should be good now.